cmd/agent/app/flare.go

```diff
@@ -116,42 +137,75 @@ func makeFlare(caseID string) error {
 	return nil
 }
 
-func requestArchive(logFile string) (string, error) {
+func requestArchive(logFile string, profile *flare.Profile) (string, error) {
 	fmt.Fprintln(color.Output, color.BlueString("Asking the agent to build the flare archive."))
 	var e error
 	c := util.GetClient(false) // FIX: get certificates right then make this true
 	ipcAddress, err := config.GetIPCAddress()
 	if err != nil {
 		fmt.Fprintln(color.Output, color.RedString(fmt.Sprintf("Error getting IPC address for the agent: %s", err)))
-		return createArchive(logFile)
+		return createArchive(logFile, profile)
 	}
+
 	urlstr := fmt.Sprintf("https://%v:%v/agent/flare", ipcAddress, config.Datadog.GetInt("cmd_port"))
 
 	// Set session token
 	e = util.SetAuthToken()
 	if e != nil {
 		fmt.Fprintln(color.Output, color.RedString(fmt.Sprintf("Error: %s", e)))
-		return createArchive(logFile)
+		return createArchive(logFile, profile)
+	}
+
+	b := bytes.NewBuffer([]byte{})
+	if profile != nil {
+		if err := json.NewEncoder(b).Encode(profile); err != nil {
+			fmt.Fprintln(color.Output, color.RedString(fmt.Sprintf("Error while encoding profile: %s", e)))
```

Comment:
We should stop and exit if we can't serialize the profile. We don't want to send half baked flare.